### PR TITLE
feat(js-auth): add MFA documentation

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes:

- adds MFA code snippets.
- aligns documentation with the native platforms.

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [x] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib-v1/auth/js/mfa_preference/update_mfa_preference.mdx [hunk 1]
```diff
@@ -0,0 +1,9 @@
+```ts
+async function handleUpdateMFAPreference() {
```
Comment (jimblanc) [inline src/fragments/lib-v1/auth/js/mfa_preference/update_mfa_preference.mdx:2]:
```suggestion
import { updateMFAPreference } from 'aws-amplify/auth';

async function handleUpdateMFAPreference() {
```

File: src/fragments/lib-v1/auth/js/sms/confirm_sign_in.mdx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+```
+
+</Block>
+
+<Block name="JavaScript">
+
+```js
+import { confirmSignIn } from 'aws-amplify/auth';
+
+async function handleConfirmSignIn(otpCode) {
```
Comment (jimblanc) [inline src/fragments/lib-v1/auth/js/sms/confirm_sign_in.mdx:23]:
```suggestion
async function handleOTPCode(otpCode) {
```

File: src/fragments/lib-v1/auth/js/sms/update_mfa_preference.mdx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+```ts
+import { updateMFAPreference } from 'aws-amplify/auth';
+
+async function handleUpdateMFAPreference() {
```
Comment (jimblanc) [inline src/fragments/lib-v1/auth/js/sms/update_mfa_preference.mdx:4]:
```suggestion
async function handleMFAPreferenceSelection() {
```

**Review by jimblanc — COMMENTED** [review]
Looking good, dropped some comments around structure & nit-picks on formatting/naming

File: src/fragments/lib-v1/auth/js/sms/sign_up.mdx [hunk 4]
```diff
@@ -5,21 +5,26 @@
...
-          phone_number: "+18880008888",
-          email: "email@email.com",
-        },
-      },
-    });
-  } catch (error) {
-    console.log(error);
-  }
+async function handleSignUp(
+	username: string,
```
Comment (jimblanc) [inline src/fragments/lib-v1/auth/js/sms/sign_up.mdx:9]: Actually I think we need to use 2-space indentation everywhere as per the style guide: https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md#21-use-2-space-indents-for-code-blocks Could you apply throughout this PR? Apologies, should have been more specific originally

**Review by jimblanc — COMMENTED** [review]
Looks good, just need to standardize the indentation

File: src/fragments/lib-v1/auth/js/sms/sign_up.mdx (hunk 4 above)
Comment (israx) [reply src/fragments/lib-v1/auth/js/sms/sign_up.mdx:9]: That's how prettier formatted the code.

Comment (israx) [reply src/fragments/lib-v1/auth/js/sms/sign_up.mdx:9]: I'm updating the PR with the new indentation

**Review by jimblanc — APPROVED** [review]

**Pull request merged**